core/server/models/plugins/filter.js

```diff
@@ -1,210 +1,226 @@
-                // TODO improve on the group by handling
-                options.groups = options.groups || [];
-                options.groups.push('posts.id');
-            }
-
-            if (joinTables && joinTables.indexOf('authors') > -1) {
-                // We need to use leftOuterJoin to insure we still include posts which don't have tags in the result
-                // The where clause should restrict which items are returned
-                this
-                    .query('leftOuterJoin', 'posts_authors', 'posts_authors.post_id', '=', 'posts.id')
-                    .query('leftOuterJoin', 'users as authors', 'posts_authors.author_id', '=', 'authors.id');
-
-                // We need to add a group by to counter the double left outer join
-                // TODO improve on the group by handling
-                options.groups = options.groups || [];
-                options.groups.push('posts.id');
-            }
+        applyDefaultAndCustomFilters: function applyDefaultAndCustomFilters(options) {
+            const nql = require('@nexes/nql');
 
-            /**
-             * @deprecated: `author`, will be removed in Ghost 3.0
-             */
-            if (joinTables && joinTables.indexOf('author') > -1) {
-                this
-                    .query('join', 'users as author', 'author.id', '=', 'posts.author_id');
-            }
-        },
+            const custom = options.filter;
+            const defaults = this.defaultFilters(options);
+            const enforced = this.enforcedFilters(options);
+            const extra = this.extraFilters(options);
 
-        /**
-         * ## fetchAndCombineFilters
-         * Helper method, uses the combineFilters util to apply filters to the current model instance
-         * based on options and the set enforced/default filters for this resource
-         * @param {Object} options
-         * @returns {Bookshelf.Model}
-         */
-        fetchAndCombineFilters: function fetchAndCombineFilters(options) {
-            options = options || {};
+            debug('custom', custom);
+            debug('extra', extra);
+            debug('default', defaults);
+            debug('enforced', enforced);
 
-            this._filters = filterUtils.combineFilters(
-                this.enforcedFilters(options),
-                this.defaultFilters(options),
-                options.filter,
-                this.extraFilters(options)
-            );
+            const filter = filterUtils.mergeFilters({enforced, defaults, custom, extra});
 
-            return this;
-        },
+            debug('filter', filter);
 
-        /**
-         * ## Apply Filters
-         * Method which makes the necessary query builder calls (through knex) for the filters set
-         * on this model instance
-         * @param {Object} options
-         * @returns {Bookshelf.Model}
-         */
-        applyDefaultAndCustomFilters: function applyDefaultAndCustomFilters(options) {
```

Comment:
Sexy Alarm: Maybe we can later rename `applyDefaultAndCustomFilters` to simply `applyFilters`.